docs/design/feature-attributes.md

```diff
@@ -0,0 +1,181 @@
+
+Properties and events declared in a class or struct with a _feature requirement_ are also in a _feature available_ scope.
+
+Lambdas and local functions inside of a method in a _feature available_ scope are also in a _feature available_ scope.
+
+Note that nested types declared in a type that is in a _feature available_ scope are not necessarily in a _feature available_ scope.
+
+## Feature requirement
+
+### Methods
+
+When `RequiresFeature` is used on a method or constructor (except static constructors), this declares a _feature requirement_ for the method.
+
+Static constructors never have a _feature requirement_. `RequiresFeature` on a static constructor is not supported.
+
+### Classes
+
+When `RequiresFeature` is used on a class, this declares a _feature requirement_ for the class.
+
+When a class has a _feature requirement_, this creates a _feature requirement_ for the following members of the class:
+  - static methods
+  - static fields
+  - all constructors (static and instance)
+  - static properties
+  - static events
+
+Note that this does not create a _feature requirement_ for nested types.
+Note also that this may create a _feature requirement_ for fields, properties, and events, which cannot have `RequiresFeature` used on them directly.
+
+### Structs
+
+When a struct has a _feature requirement_, this creates a _feature requirement_ for the following members of the struct:
+  - all methods
+  - all constructors (static and instance)
+  - all fields
+  - all properties
+  - all events
+
+Note that structs may have _feature requirement_ due to compiler-generated code, even though they can not have `RequiresFeature`.
+
+### State machine types
+
+When an iterator or async method is in a _feature available_ scope, the compiler-generated state machine class or type has a _feature requirement_.
+
+### Nested functions
+
+When a lambda or local function is declared in a method which is in a _feature available_ scope, then the following compiler-generated type or members have a _feature requirement_:
+
+- The generated closure environment type, if it is unique to the lambda or local function, OR
+
+- The generated method for the lambda or local function, if the compiler does not generate a type for the closure environment, OR
+
+- The generated method and delegate cache field for the lambda or local function, if these are generated into a static closure environment type.
+
+For analyzers which don't have visibility into the compiler-generated code for nested functions, nested functions declared in a method which is in a _feature available_ scope have a _feature requirement_.
+
+Note that a lambda or local function inherits _feature requirement_ from the enclosing user method, not from an enclosing lambda or local function if one is present.
+
+## Validation behavior
+
+### RequiresFeatureAttribute
+
+`RequiresFeatureAttribute` on a static constructor is disallowed.
```

Comment:
Above you use just `RequiresFeature` (without the "Attributes").... just seems inconsistent.